docker/tests/data-sync/build_image.sh

```
@@ -5,18 +5,29 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
+BUILD_ARGS=""
+if [ "$ENABLE_TELEMETRY" = "true" ] || [ "$ENABLE_TELEMETRY" = "1" ]; then
+    BUILD_ARGS="--build-arg CARGO_FEATURES=telemetry"
+    echo "Building with telemetry support..."
+else
+    echo "Building without telemetry (set ENABLE_TELEMETRY=true to enable)"
+fi
+
 # Use DOCKER_BUILDKIT=0 to avoid buildx issues
 DOCKER_BUILDKIT=0 docker build \
+    $BUILD_ARGS \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)